### PR TITLE
[TT-6473] Add hybrid pump to hybrid chart

### PR DESCRIPTION
**Pull request by joshblakeley**
## Description

Current hybrid helm chart lacks the ability to natively run pump when you need it alongside your hybrid gateway.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by caroltyk** [comment]
hi @joshblakeley thanks for raising this! Could you also supplement Test Coverage For This Change and provide any related docs update please?

**Comment by letzya** [comment]
@caroltyk @joshblakeley this of course will be possible in the new v1.X release with pump being a subchart

File: tyk-hybrid/templates/deployment-pump.yaml [hunk 1]
```diff
@@ -23,106 +23,100 @@ spec:
...
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_DATABASE
+            value: "{{ .Values.redis.storage.database }}"
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_REDISUSESSL
+            value: "{{ default "false" .Values.redis.useSSL }}"
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_ADDRS
+            value: {{ include "tyk-hybrid.redis_url" . | quote }}
+          - name: TYK_PMP_ANALYTICSSTORAGECONFIG_ENABLECLUSTER
+            value: "{{ default "false" .Values.redis.enableCluster }}"
+          - name: TYK_PMP_PUMPS_HYBRID_TYPE
+            value: "{{ default "hybrid" .Values.pump.type }}"
```
Comment (zalbiraw) [inline tyk-hybrid/templates/deployment-pump.yaml:68]: Why are we adding the ability to change the pump type? if someone changes that why do we add the rest of the hybrid pump config?

File: tyk-hybrid/templates/deployment-pump.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          - name: TYK_PMP_PUMPS_HYBRID_META_USESSL
+            value: "{{ default "true" .Values.pump.useSSL }}"
+          - name: TYK_PMP_PUMPS_HYBRID_META_SSLINSECURESKIPVERIFY
+            value: "true"
```
Comment (komalsukhani) [inline tyk-hybrid/templates/deployment-pump.yaml:72]: Shouldn't it be set to "false" by default?

**Comment by caroltyk** [comment]
This feature is enabled in our new helm chart for Tyk Data Plane:
https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-mdcb-data-plane

**Comment by zalbiraw** [comment]
Added to new charts

**Pull request closed without merging**